File: src/main/java/be/superjoran/mint/domain/CategoryType.java
package be.superjoran.mint.domain;

/**
 * Created by Jorandeboever on 5/7/2017.
 */
public enum CategoryType implements Display {
    INCOME, EXPENSE, UNTRACKED;

    @Override
    public String getId() {
        return this.name();
    }

    @Override
    public String getDisplayValue() {
        return this.name();
    }
}

File: src/main/java/be/superjoran/mint/domain/CategoryType.kt
package be.superjoran.mint.domain

/**
 * Created by Jorandeboever on 5/7/2017.
 */
enum class CategoryType : Display {
    INCOME, EXPENSE, UNTRACKED;

    override val id: String?
        get() = this.name

    override val displayValue: String?
        get() = this.name
}
